pkg/kubectl/cmd/version.go

```diff
@@ -46,36 +54,74 @@ func NewCmdVersion(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	}
 	cmd.Flags().BoolP("client", "c", false, "Client version only (no server required).")
 	cmd.Flags().BoolP("short", "", false, "Print just the version number.")
+	cmd.Flags().String("output", "", "output format, options available are yaml and json")
 	cmd.Flags().MarkShorthandDeprecated("client", "please use --client instead.")
 	return cmd
 }
 
 func RunVersion(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
-	v := fmt.Sprintf("%#v", version.Get())
-	if cmdutil.GetFlagBool(cmd, "short") {
-		v = version.Get().GitVersion
+	var serverVersion *apimachineryversion.Info = nil
+	var serverErr error = nil
+	vo := VersionObj{nil, nil}
+
+	clientVersion := version.Get()
+	vo.ClientVersion = &clientVersion
+
+	if !cmdutil.GetFlagBool(cmd, "client") {
+		serverVersion, serverErr = retrieveServerVersion(f)
+		vo.ServerVersion = serverVersion
 	}
 
-	fmt.Fprintf(out, "Client Version: %s\n", v)
-	if cmdutil.GetFlagBool(cmd, "client") {
-		return nil
+	switch of := cmdutil.GetFlagString(cmd, "output"); of {
+	case "":
+		if cmdutil.GetFlagBool(cmd, "short") {
+			fmt.Fprintf(out, "Client Version: %s\n", clientVersion.GitVersion)
+
+			if serverVersion != nil {
+				fmt.Fprintf(out, "Server Version: %s\n", serverVersion.GitVersion)
+			}
+		} else {
+			fmt.Fprintf(out, "Client Version: %s\n", fmt.Sprintf("%#v", clientVersion))
+
+			if serverVersion != nil {
+				fmt.Fprintf(out, "Server Version: %s\n", fmt.Sprintf("%#v", *serverVersion))
+			}
+		}
+	case "yaml":
+		y, err := yaml.Marshal(&vo)
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintln(out, string(y))
+	case "json":
+		y, err := json.Marshal(&vo)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(out, string(y))
+	default:
+		return errors.New("invalid output format: " + of)
+
 	}
 
-	clientset, err := f.ClientSet()
-	if err != nil {
-		return err
+	if serverErr != nil {
+		return serverErr
 	}
 
-	serverVersion, err := clientset.Discovery().ServerVersion()
+	return nil
+}
+
+func retrieveServerVersion(f cmdutil.Factory) (*apimachineryversion.Info, error) {
```

Comment:
very interesting idea, I would say to keep things clean and within scope, that we should open a new PR for that if we feel strongly about it.